test_load_balancer.py

```diff
@@ -45,6 +45,36 @@ def test_simple_tcp_load_balancer(prober, create_load_balancer_scenario):
     assert 'Backend server running on' in content
 
 
+def test_simple_udp_load_balancer(prober, create_load_balancer_scenario):
+    """ Create a simple UDP load balancer with one backend. """
+
+    # Create a load balancer setup with one backend on a private network
+    load_balancer, listener, pool, backends, private_network = \
+        create_load_balancer_scenario(
+            num_backends=1,
+            algorithm='round_robin',
+            port=80,
+            pool_protocol='udp',
+            listener_protocol='udp',
+            ssl=False,
+            health_monitor_type=None,
+            allowed_cidrs=None,
+        )
+
+    backend, = backends
+
+    def assert_udp_response():
+        assert (f'Backend server running on {backend.name} {backend.uuid}.'
+                == prober.udp_get(load_balancer.vip(4), 80))
```

Comment:
I think if we assume that `socat` is installed (and otherwise install it, if we ever have an image where that is not the case), it would be cleaner to just run the command inline:

```python
expected = f'Backend server running on {backend.name} {backend.uuid}.'
assert prober.output_of('echo -n "hi {ip}" | socat -t {timeout} - UDP:{ip}:{port}') == expected
```

The main reason for this: I like to surface shell commands as much as possible, as they are instructive for people without Python knowledge (some of our audience for these tests).

Also, what we pipe into this does not matter right? If so, we might get `socat` to send something without piping. Maybe there's something like `--data='x'` or so?

Reply:
Ah I was not aware of `http_get`. In that case I think using your function makes sense. Piping is fine with me as well in this case, as this is now mostly hidden from the target audience.